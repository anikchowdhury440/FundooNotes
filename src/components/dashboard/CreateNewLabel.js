import React, {Component} from 'react';
import {View, Text, TextInput, TouchableWithoutFeedback} from 'react-native';
import {Appbar} from 'react-native-paper';
import CreateNewLabelStyle from '../../styles/CreateNewLabel.styles'

export default class CreateNewLabel extends Component {
    constructor(props) {
        super(props)
        this.state = {
            createLabel : true
        }
    }

    handleBackIconButton = () => {
        this.props.navigation.goBack()
    }

    handleCreateLabel = () => {
        this.setState({
            createLabel : !this.state.createLabel
        })
    }

    render() {
        return(
            <View style = {CreateNewLabelStyle.mainContainer}>
                <View>
                    <Appbar style = {CreateNewLabelStyle.header_style}>
                        <Appbar.Action 
                            style = {{marginLeft : 10}}
                            icon = 'keyboard-backspace'
                            onPress = {this.handleBackIconButton}/>
                        <Appbar.Content 
                            title = 'Edit Labels'/>
                    </Appbar>
                </View>
                <View style = {this.state.createLabel ? CreateNewLabelStyle.create_label_appbar : null}>
                    <Appbar style = {CreateNewLabelStyle.header_style}>
                        {
                            (this.state.createLabel ? 
                                <Appbar.Action 
                                    style = {{marginLeft : 10}}
                                    icon = 'close'
                                    onPress = {this.handleCreateLabel}/>
                                : 
                                <Appbar.Action 
                                    style = {{marginLeft : 10}}
                                    icon = 'plus'
                                    onPress = {this.handleCreateLabel}/>)
                        }
                        <TouchableWithoutFeedback onPress = {this.handleCreateLabel}>
                            <View style = {{width : '65%'}}>
                                <TextInput
                                    style = {CreateNewLabelStyle.textinput_style}    
                                    placeholder = 'Create New Label'
                                    editable = {this.state.createLabel ? true : false}
                                    autoFocus = {true}/>
                            </View>
                        </TouchableWithoutFeedback>
                        <Appbar.Content/>
                        {
                            (this.state.createLabel ? 
                                <Appbar.Action 
                                    icon = 'check'/>
                                : 
                                null)
                        }
                    </Appbar>
                </View>
            </View>
        )
    }
}