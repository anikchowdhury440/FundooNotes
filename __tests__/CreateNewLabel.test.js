import React from 'react';
import {shallow} from 'enzyme';
import CreateNewLabel from '../src/components/dashboard/CreateNewLabel';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const initialState = {
    userId : '',
    userLabel : [],
    screenName : '',
    labelKey : ''
}
const store = mockStore(initialState)

describe('test CreateNewLabel', () => {
    it('should match to snapshot', () => {
        const component = shallow(<CreateNewLabel store = {store}/>)
        expect(component).toMatchSnapshot();
    })
})