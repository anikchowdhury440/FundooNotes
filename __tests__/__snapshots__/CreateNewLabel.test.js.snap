// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CreateNewLabel should match to snapshot 1`] = `ShallowWrapper {}`;
